### DISCUSSION TITLE
Add support for separate coinbase smart wallet

### DISCUSSION THREAD

**Discussion by KarmanMax** [rainbow-me/rainbowkit#2041] [Ideas]
There are situations both normal coinbaseWallet (preference: "eoaOnly") and coinbase smart wallet ((preference: "smartWalletOnly") being required in the wallet list.

And I found the "smartWalletOnly" option of coinbaseWallet doesn't work, it still always connect to the coinbase extension when the coinbase extension is enabled.

**Comment by DanielSinclair** [comment]
Hey @KarmanMax! We would like to separate these concepts as well and include our own connectivity logic in RainbowKit, but Coinbase currently controls this flow entirely in their SDK, so I'd recommend you relay feedback to them directly here: https://github.com/coinbase/coinbase-wallet-sdk
